executor/slow_query.go

```diff
@@ -46,34 +46,32 @@ import (
 type slowQueryRetriever struct {
 	table       *model.TableInfo
 	outputCols  []*model.ColumnInfo
-	retrieved   bool
 	initialized bool
 	extractor   *plannercore.SlowQueryExtractor
 	files       []logFile
 	fileIdx     int
 	fileLine    int
 	checker     *slowLogChecker
+
+	parsedSlowLogCh chan parsedSlowLog
+	test            bool
```

Comment:
The variable of `test` is so weird, Please don't write the special logic for the test.

Reply:
Sorry about that. It's a quick solution to make existed unit test in `testExecSuite.TestSlowQueryRetriever` passed because we changed the `initialized()` behavior.

Reply:
Do we have other solution to fix this?

Reply:
If we don't write special logic for testing, we have to change the unit test. Because our optimization changes are for the executors, which will only call `retrieve()`. A solution could be changing `testExecSuite.TestSlowQueryRetriever` to only rely on the `retrieve()` method.

Reply:
Great, How about change the test logic and remove this variable.

Reply:
I split the asynchronous parsing goroutine starting out of `initialize()` and add it to `retrieve()`. Now we won't affect the original unit test because we only change the `retrieve()` behavior, which will be utilized by executors for optimization.